client/rpmtrans.c

```diff
@@ -472,6 +472,86 @@ TDNFTransAddReInstallPkgs(
     return TDNFTransAddInstallPkgs(pTS, pTdnf, pInfo);
 }
 
+uint32_t
+TDNFFetchRemoteGPGKey(
+    PTDNF pTdnf,
+    const char* pszRepoName,
+    const char* pszUrlGPGKey,
+    char** ppszKeyLocation)
+{
+    uint32_t dwError = 0;
+    char* pszKeyCacheDir = NULL;
+    char* pszFilePath = NULL;
+    char* pszFilePathCopy = NULL;
+    char* pszDownloadCacheDir = NULL;
+    char* pszKeyLocation = NULL;
+
+    if(!pTdnf || IsNullOrEmptyString(pszRepoName || IsNullOrEmptyString(pszUrlGPGKey)))
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = TDNFAllocateStringPrintf(
+                  &pszKeyCacheDir,
+                  "%s/%s/%s",
+                  pTdnf->pConf->pszCacheDir,
+                  pszRepoName,
+                  "keys");
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    dwError = TDNFFileNameFromUri(pszUrlGPGKey, &pszKeyLocation);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    dwError = TDNFAllocateStringPrintf(
+                  &pszFilePath,
+                  "%s/%s",
+                  pszKeyCacheDir,
```

Comment:
Looks like this is the only place where `pszKeyCacheDir` is being used. Can we from `pszFilePath` directly?

Reply:
Good point. Fixed.